src/MySqlConnector/Core/Row.cs

```diff
@@ -393,17 +370,17 @@ public int GetValues(object[] values)
 
 	public object this[string name] => GetValue(ResultSet.GetOrdinal(name));
 
-	protected Row(ResultSet resultSet)
+	protected Row(bool binary, ResultSet resultSet)
 	{
 		ResultSet = resultSet;
 		m_dataOffsets = new int[ResultSet.ColumnDefinitions!.Length];
 		m_dataLengths = new int[ResultSet.ColumnDefinitions.Length];
+		columnReaders = Array.ConvertAll(ResultSet.ColumnDefinitions,
+			new Converter<ColumnDefinitionPayload, IColumnReader>(column => ColumnReaderFactory.GetReader(binary, column, resultSet.Connection)));
```

Comment:
Using `Array.ConvertAll` and a delegate allocation seems needlessly inefficient compared to a `foreach` loop.

Will fix in https://github.com/mysql-net/MySqlConnector/pull/1332/commits/3e48b93214d105b06e13b88f1531e471dc50ba10.